lib/trie/database.go

```diff
@@ -23,6 +23,12 @@ var ErrEmptyProof = errors.New("proof slice empty")
 // and the value is the encoded node.
 // Generally, this will only be used for the genesis trie.
 func (t *Trie) Store(db chaindb.Database) error {
+	for _, v := range t.childTries {
+		if err := v.Store(db); err != nil {
+			return err
```

Comment:
Please wrap the error with some context information such as the child trie root hash (key in t.childTries)